src/Collection/CollectionTrait.php

```diff
@@ -876,6 +876,7 @@ public function chunkWithKeys(int $chunkSize, bool $keepKeys = true): Collection
      */
     public function isEmpty(): bool
     {
+        // phpcs:ignore SlevomatCodingStandard.Variables.UnusedVariable.UnusedVariable
         foreach ($this as $el) {
```

Comment:
Is there no way to just do an empty check?

Reply:
Since CollectionInterface extends Iterator perhaps we could do a rewind() followed by valid() check?